### DISCUSSION TITLE
How to update docker applications?

### DISCUSSION THREAD

**Discussion by sinoserendipity** [IceWhaleTech/CasaOS#744] [General]
I find there is no way to update docker applications! it's important to keep updated!

**Comment by cod-e** [comment]
I am having an issue as well attempting to update the casa os. It gives me the alert message, prompting the update. When I click through to engage the update. The update pop up appears, then when clicking the blue update button. The button begins to show busy prompt, but nothing happens. The page just stays the same. After an hour of seeing it just remain idle. I refresh the page and the casa ui returns to home page. Is there a work around for this?

**Comment by delyan** [comment]
Update of the installed apps really breaks the promise of what CasaOS is aiming to accomplish.
We see each app installed from the store is bound to a specific release, probably because the support for updating docker images is missing. But that makes the whole system pretty static, which is a "No".
Please consider support for proper docker images update (not automatic, though).
Also CasaOS self-update is almost missing, the scripts provided to run from the terminal are not well tested and buggy.

**Reply by delyan** [reply]
Just to describe how I "hack" it to update an app:
1. Export the app (docker compose) from it's settings.
2. Edit the yaml file, adding "latest" to the docker image URL
3. Stop the app that will be updated.
4. Install a custom app from the appstore by uploading the edited yaml file.
5. Start the update app - it actually replaces (updates) the older docker app image, providing all volumes are correctly set before the procedure.

**Reply by TrancaDev** [reply]
This workaround is great. Thanks!

PS: I ended up installing Portainer which lets me change the images and recreating the container more easily :)
